plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestS3WithDoubleSlashPathAndS3ASchemeTest.java

```diff
@@ -0,0 +1,117 @@
+        return IcebergQueryRunner.builder()
+                .setIcebergProperties(ImmutableMap.<String, String>builder()
+                        .put("iceberg.catalog.type", "TESTING_FILE_METASTORE")
+                        .put("hive.metastore.catalog.dir", "s3://%s/".formatted(bucketName))
+                        .put("hive.s3.aws-access-key", "accesskey")
+                        .put("hive.s3.aws-secret-key", "secretkey")
+                        .put("hive.s3.endpoint", LOCALSTACK.getEndpointOverride(LocalStackContainer.Service.S3).toString())
+                        .put("hive.s3.region", Region.of(LOCALSTACK.getRegion()).toString())
+                        .put("fs.hadoop.enabled", "true")
+                        .put("fs.native-s3.enabled", "false")
+                        .buildOrThrow())
+                .setCatalog2IcebergProperties(ImmutableMap.<String, String>builder()
+                        .put("iceberg.catalog.type", "TESTING_FILE_METASTORE")
+                        .put("hive.metastore.catalog.dir", "s3://%s/".formatted(bucketName))
+                        .put("s3.aws-access-key", "accesskey")
+                        .put("s3.aws-secret-key", "secretkey")
+                        .put("s3.endpoint", LOCALSTACK.getEndpointOverride(LocalStackContainer.Service.S3).toString())
+                        .put("s3.region", Region.of(LOCALSTACK.getRegion()).toString())
+                        .put("fs.hadoop.enabled", "false")
+                        .put("fs.native-s3.enabled", "true")
+                        .buildOrThrow())
+                .build();
+    }
+
+    @Test
+    public void testS3ASchemeTableInHadoopFsAndInNativeFs()
+    {
+        String schemaName = "test_schema";
+        String tableName = "sample_table";
+        String level1 = "level1";
+        String level2 = "level2";
+
+        assertUpdate("CREATE SCHEMA %s.%s".formatted(ICEBERG_CATALOG, schemaName));
+        // Create table in HadoopFs
+        assertUpdate("CREATE TABLE %s.%s.%s WITH (location = 's3a://%s/%s/%s//%s') AS SELECT 'test value' col".formatted(ICEBERG_CATALOG, schemaName, tableName, bucketName, level1, level2, tableName), 1);
+        // Verify that table is created at correct location
+        assertThat(s3.listObjectsV2(builder -> builder.bucket(bucketName).prefix("%s/%s//%s/".formatted(level1, level2, tableName))).contents().isEmpty()).isFalse();
+        assertThat(s3.listObjectsV2(builder -> builder.bucket(bucketName).prefix("%s/%s/%s/".formatted(level1, level2, tableName))).contents().isEmpty()).isTrue();
+
+        // Query table in HadoopFs
+        assertQuery("SELECT * FROM %s.%s.%s".formatted(ICEBERG_CATALOG, schemaName, tableName), "VALUES 'test value'");
+
+        // Query the same table in NativeFs
+        assertQuery("SELECT * FROM %s.%s.%s".formatted(ICEBERG_CATALOG2, schemaName, tableName), "VALUES 'test value'");
+
+        assertUpdate("DROP TABLE %s.%s.%s".formatted(ICEBERG_CATALOG, schemaName, tableName));
```

Comment:
You can extract this code to a separate method which receives the location of the table and you can check this way the compatibility for both canonical as well as non-canonical paths between using s3 hadoop & s3 native.

In this way we can have full coverage for s3 / s3a/ s3n locations to avoid (although unlikely given the deprecation of hadoop fs s3 layer) potential regressions.

Reply:
Yeah good idea, added test covering canonical and non-canonical paths with different schemes.